src/core/core.scale.js

```diff
@@ -305,7 +305,13 @@ function skip(ticks, spacing, majorStart, majorEnd) {
 	}
 
 	next = start;
-	for (i = Math.max(start, 0); i < end; i++) {
+
+	while (next < 0) {
+		count++;
+		next = Math.round(start + count * spacing);
+	}
+
+	for (i = next; i < end; i++) {
```

Comment:
is it ok that ticks between 0 and `next` don't get skipped for first segment (assuming next < 0 to start with)?

Reply:
good catch. I've updated it